manim/scene/scene_file_writer.py

```diff
@@ -439,6 +443,25 @@ def close_movie_pipe(self):
             self.temp_partial_movie_file_path,
             self.partial_movie_file_path,
         )
+        logger.debug(
+            f"Animation {self.scene.num_plays} : Partial movie file written in {self.partial_movie_file_path}")
+
+    def is_already_cached(self, hash_invokation):
+        """Will check if a file named with `hash_invokation` exists.
+
+        Parameters
+        ----------
+        hash_invokation : :class:`str`
+            The hash corresponding to an invokation to either `scene.play` or `scene.wait`.
+
+        Returns
+        -------
+        `bool`
+            Wether the file exists.
+        """
+        path = os.path.join(self.partial_movie_directory, "{}{}".format(
+            hash_invokation, self.movie_file_extension))
+        return os.path.exists(path)
```

Comment:
Here is a bug. 
If you use `-s` argument in cli, it will save the last frame, but notice at line 74 to 108 in `scene_file_writer.py`, this will not generate attribute `self.partial_movie_directory`, so it throw:
```
AttributeError: 'SceneFileWriter' object has no attribute 'partial_movie_directory'
```
To fix this, I think we can just let this function return `False` when `self.write_to_movie == False` , because save last frame of a scene doesn't need to use cache.

Reply:
And I think this may be the reason why the test was failed.

Reply:
I added to the CONFIG of the tests `enable_caching = False`. It still does not work, but now it is a different bug related to the JSON encoder. I will fix that in a few days

Reply:
But I think it still needs to be changed here or other related places, otherwise `-s` must need to let `enable_caching=False`, which is a bit inconvenient.

Reply:
I don't understand your point. What do you want to do with `-s` ? My idea was that `-s` would automatically disable `enable_caching`, is that what you meant ?

Because the tests don't use flag to be ran. They use a custom CONFIG, (see `tests/conftest.py`), so `-s` is not related to the test failing. 
And concerning the bug, I see, that's an easy fix. When write_to_movie == False, i will init `self.partial_movie_directory` to `None`.

Reply:
Yes, I mean that when you use `-s` to save the last frame, an exception will be thrown and it will not work. 
My main purpose is not to solve the problem that the test fails, but if this is not changed, then the `-s` arg will be broken.

Reply:
This should be fixed now. Feel free to check :)